discovery-infra/utils.py

```diff
@@ -255,6 +286,14 @@ def wait_till_cluster_is_in_status(
         log.info("Cluster: %s", client.cluster_get(cluster_id))
         raise
 
+def is_cluster_in_status(client, cluster_id, statuses):
+    log.info("Is cluster %s in status %s", cluster_id, statuses)
+    try:
+        return client.cluster_get(cluster_id).status in statuses
+    except:
+        log.info("Cluster: %s", client.cluster_get(cluster_id))
```

Comment:
all u do in try is to get cluster. So if it fails there u want one more time to run it?
I think u need something like this .
log.exception("Failed to get cluster")

Reply:
So, log the exception and try once more? sure

Reply:
done